shared/src/main/scala/tastyquery/ast/Types.scala

```diff
@@ -102,8 +102,17 @@ object Types {
           else
             tycon match
               case tycon: Symbolic if tycon.symbol.isClass =>
-                // Fast path
-                ClassRef(tycon.symbol.asClass)
+                val cls = tycon.symbol.asClass
+                if cls.isDerivedValueClass then
```

Comment:
You'll also have to perform this check in the `case tpe: Symbolic` case, for monomorphic value classes.